src/MakePotCommand.php

```diff
@@ -147,38 +151,50 @@ public function handle_arguments( $args, $assoc_args ) {
 			}
 
 			$this->domain = Utils\get_flag_value( $assoc_args, 'domain', $this->domain );
+
+			WP_CLI::debug( sprintf( 'Extracting all strings with text domain "%s"', $this->domain ), 'make-pot' );
 		}
 
 		// Determine destination.
 		$this->destination = "{$this->source}/{$this->slug}.pot";
 
 		if ( ! empty( $file_data['Domain Path'] ) ) {
 			// Domain Path inside source folder.
-			$this->destination = sprintf( '%s/%s/%s.pot', $this->source, $file_data['Domain Path'], $this->slug );
+			$this->destination = sprintf(
+				'%s/%s/%s.pot',
+				$this->source,
+				$this->unslashit( $file_data['Domain Path'] ),
+				$this->slug
+			);
 		}
 
 		if ( isset( $args[1] ) ) {
 			$this->destination = $args[1];
 		}
 
+		WP_CLI::debug( sprintf( 'Destination: %s', $this->destination ), 'make-pot' );
+
 		// Two is_dir() checks in case of a race condition.
-		if ( ! is_dir( dirname( $this->destination ) ) && ! mkdir( dirname( $this->destination ) ) && ! is_dir( dirname( $this->destination ) ) ) {
+		if ( ! is_dir( dirname( $this->destination ) ) &&
+		     ! mkdir( dirname( $this->destination ), 0777, true ) &&
+		     ! is_dir( dirname( $this->destination ) )
+		) {
 			WP_CLI::error( 'Could not create destination directory!' );
 		}
 
 		if ( isset( $assoc_args['merge'] ) ) {
-			if ( true === $assoc_args['merge'] && file_exists( $this->destination ) ) {
+			if ( true === $assoc_args['merge'] ) {
 				$this->merge = $this->destination;
 			} elseif ( ! empty( $assoc_args['merge'] ) ) {
-				if ( ! file_exists( $assoc_args['merge'] ) ) {
-					WP_CLI::error( sprintf( 'Invalid file %s', $assoc_args['merge'] ) );
-				}
-
 				$this->merge = $assoc_args['merge'];
 			}
-		}
 
-		WP_CLI::debug( sprintf( 'Destination: %s', $this->destination ), 'make-pot' );
+			if ( isset( $this->merge ) && ! file_exists( $this->merge ) ) {
+				WP_CLI::warning( sprintf( 'POT file to merge with does not exist: %s', $this->merge ) );
```

Comment:
Typo: `with` => `which`

Reply:
Ah, I read it in a different way: "Here is a reference to the POT file we wanted to merge but _which_ does not exist: ...".

Some random ideas for an alternative...

* "Merge target does not exist: ..."
* "POT file to be merged does not exist: ..."
* "Invalid POT file set as merge target: ..."
* "File to merge is not valid: ..."
* "Invalid file provided to --merge: ..."